### PR TITLE
[data] move many aggregate functions internal

### PR DESCRIPTION
**Pull request by can-anyscale**
Move most of the aggregate functions internal (except for `AggragateFn`), as discussed with @bveeramani 

Test:
- CI

### CODE REVIEW

**Review by bveeramani — APPROVED** [review]

File: python/ray/data/_internal/aggregate/_aggregate.py
Comment (can-anyscale) [reply python/ray/data/_internal/aggregate/_aggregate.py:1]: nice, let's do that

**Comment by can-anyscale** [comment]
made ray.data._internal.aggregate a module

**Comment by can-anyscale** [comment]
need a few more code owner approval @justinvyu , @matthewdeng , @hongpeng-guo , thankkks

**Review by matthewdeng — APPROVED** [review]

**Pull request merged**
